argus/action_manager/windows.py

```diff
@@ -283,6 +286,50 @@ def is_dir(self, path):
         """
         return self._exists(path, self.PATH_CONTAINER)
 
+    def _new_item(self, path, item_type):
+        """Create a New-Item.
+
+        :param path:
+            Instance path of the new item.
+        :param item_type:
+            It can be `Directory` or `File`
+        """
+        cmd = "New-Item -Path {} -Type {} -Force".format(path, item_type)
+        self._client.run_command_with_retry(cmd=cmd,
+                                            command_type=util.POWERSHELL)
+
+    def mkdir(self, path):
+        """Create a directory in the instance if the path is valid.
+
+        :param path:
+            Remote path where the new item should be created.
+        """
+        if self.is_dir(path):
+            cmd = ("$datetime = get-date;"
+                   "$dir = Get-Item '{}';"
+                   "$dir.CreationTime = $datetime;"
+                   "$dir.LastWriteTime = $datetime;"
+                   "$dir.LastAccessTime = $datetime;").format(path)
+            self._client.run_command_with_retry(
+                cmd, command_type=util.POWERSHELL)
+            return
+
+        self._new_item(path, self._DIRECTORY)
+
+    def mkfile(self, path):
+        """Create a file in the instance if the path is valid.
+
+        :param path:
+            Remote path where the new item should be created.
+        """
+        if self.exists(path):
```

Comment:
Also, you should test if path refers to a directory or file, because if it is a directory, your Powershell command will fail.